doc/lua_api.txt

```diff
@@ -3300,6 +3300,18 @@ Player Inventory lists
     * Is not created automatically, use `InvRef:set_size`
     * Is only used to enhance the empty hand's tool capabilities
 
+ItemStack transation order
+--------------------------
+
+This list assumes non-empty ItemStacks in both slots (swapping).
+Put/take callbacks about empty ItemStack are not executed.
+
+1. The "allow take" and "allow put" callbacks are each run once for the source
```

Comment:
I would split it into two lines.
1. The "allow take" callbacks are run for the source and destination inventory.
2. The "allow put" callbacks are run for the source and destination inventory.
And if we want to fix the order between the source and destination inventory: "run for the source and then for the destination inventory".

Reply:
Actually the `allow_take`/`allow_put` order does not matter because no items are actually moved. It is simply to determine them movable item count.